<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1920</width>
    <height>1080</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="7,3">
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="page">
       <property name="font">
        <font>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QGraphicsView" name="graphicsView_Mark"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_2">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="ExtendQGraphicsView" name="graphicsView_Camera"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QStackedWidget" name="stackedWidget_Parameters">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="page_3">
       <widget class="QLabel" name="label_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>29</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>mark1</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="page_4">
       <widget class="QTabWidget" name="tabWidget">
        <property name="geometry">
         <rect>
          <x>40</x>
          <y>490</y>
          <width>401</width>
          <height>281</height>
         </rect>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="tabPosition">
         <enum>QTabWidget::East</enum>
        </property>
        <property name="tabShape">
         <enum>QTabWidget::Triangular</enum>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab_1">
         <attribute name="title">
          <string>Image Process</string>
         </attribute>
         <widget class="QGroupBox" name="groupBox">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>20</y>
            <width>321</width>
            <height>121</height>
           </rect>
          </property>
          <property name="title">
           <string>Threshold</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <item>
                <widget class="QCheckBox" name="checkBox_AutoThreshold">
                 <property name="text">
                  <string>Auto</string>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">buttonGroup</string>
                 </attribute>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="checkBox_ManualThreshold">
                 <property name="text">
                  <string>Manual</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">buttonGroup</string>
                 </attribute>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="2,6,2">
               <item>
                <widget class="QSpinBox" name="spinBox_LeftThreshold">
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                 <property name="maximum">
                  <number>254</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QxtSpanSlider" name="horizontalSlider_Threshold">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinBox_RightThreshold">
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Calibration</string>
         </attribute>
        </widget>
        <widget class="QWidget" name="tab_3">
         <attribute name="title">
          <string>Egde Finder</string>
         </attribute>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_CameraInfo">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>60</y>
          <width>351</width>
          <height>111</height>
         </rect>
        </property>
        <property name="title">
         <string>Camera Info</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QComboBox" name="comboBox_CameraInfo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QCheckBox" name="checkBox_AutoExposure">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>AutoExposure</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QLabel" name="label">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Exposure:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="doubleSpinBox_Exposure">
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="decimals">
                <number>1</number>
               </property>
               <property name="maximum">
                <double>99999999.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Frame Rate:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_FrameRate">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeType">
                <enum>QSizePolicy::Expanding</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QPushButton" name="pushButton_ConnectedCamera">
             <property name="text">
              <string>Connected</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_DisConnectedCamera">
             <property name="text">
              <string>DisConnected</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="geometry">
         <rect>
          <x>170</x>
          <y>830</y>
          <width>171</width>
          <height>91</height>
         </rect>
        </property>
        <property name="title">
         <string>Motion</string>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox_LoadImage">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>180</y>
          <width>351</width>
          <height>58</height>
         </rect>
        </property>
        <property name="title">
         <string>Load Image</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_11">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QLineEdit" name="lineEdit_ImageFilePath">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_LoadImage">
             <property name="text">
              <string>load Image</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1920</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar_Tool">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpenFile"/>
   <addaction name="actionSaveFile"/>
   <addaction name="separator"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="separator"/>
   <addaction name="actionOpenMarkInterface"/>
   <addaction name="actionOpenCameraInterface"/>
  </widget>
  <widget class="QToolBar" name="toolBar_Shapes">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>BottomToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSelect"/>
   <addaction name="actionDrawLine"/>
   <addaction name="actionDrawRectangle"/>
   <addaction name="actionDrawRoundedCornerRectangle"/>
   <addaction name="actionDrawTriangle"/>
   <addaction name="actionDrawCircle"/>
   <addaction name="actionDrawEllipse"/>
   <addaction name="actionDrawPolygon"/>
  </widget>
  <widget class="QToolBar" name="toolBar_Camera">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionGrabOnce"/>
   <addaction name="actionGrabContinue"/>
   <addaction name="actionGrabStop"/>
  </widget>
  <action name="actionOpenFile">
   <property name="icon">
    <iconset>
     <normaloff>ico/open_icon.png</normaloff>ico/open_icon.png</iconset>
   </property>
   <property name="text">
    <string>OpenFile</string>
   </property>
   <property name="toolTip">
    <string>open file from the disk</string>
   </property>
  </action>
  <action name="actionSaveFile">
   <property name="icon">
    <iconset>
     <normaloff>ico/save_icon.png</normaloff>ico/save_icon.png</iconset>
   </property>
   <property name="text">
    <string>SaveFile</string>
   </property>
   <property name="toolTip">
    <string>save file to disk</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset>
     <normaloff>ico/undo_back_left_icon.png</normaloff>ico/undo_back_left_icon.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="toolTip">
    <string>return last action</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset>
     <normaloff>ico/redo_forward_right_icon.png</normaloff>ico/redo_forward_right_icon.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="toolTip">
    <string>forward next action</string>
   </property>
  </action>
  <action name="actionOpenMarkInterface">
   <property name="icon">
    <iconset>
     <normaloff>ico/laser_wip_icon.png</normaloff>ico/laser_wip_icon.png</iconset>
   </property>
   <property name="text">
    <string>OpenMarkInterface</string>
   </property>
   <property name="toolTip">
    <string>open the laser marking interface</string>
   </property>
  </action>
  <action name="actionOpenCameraInterface">
   <property name="icon">
    <iconset>
     <normaloff>ico/camera_icon.png</normaloff>ico/camera_icon.png</iconset>
   </property>
   <property name="text">
    <string>OpenCameraInterface</string>
   </property>
   <property name="toolTip">
    <string>open the vision interface</string>
   </property>
  </action>
  <action name="actionSelect">
   <property name="icon">
    <iconset>
     <normaloff>ico/_selection_tool_icon.png</normaloff>ico/_selection_tool_icon.png</iconset>
   </property>
   <property name="text">
    <string>Select</string>
   </property>
   <property name="toolTip">
    <string>select a shape(no draw)</string>
   </property>
  </action>
  <action name="actionDrawLine">
   <property name="icon">
    <iconset>
     <normaloff>ico/line_icon.png</normaloff>ico/line_icon.png</iconset>
   </property>
   <property name="text">
    <string>DrawLine</string>
   </property>
   <property name="toolTip">
    <string>draw a line</string>
   </property>
  </action>
  <action name="actionDrawRectangle">
   <property name="icon">
    <iconset>
     <normaloff>ico/rectangle_shape_icon.png</normaloff>ico/rectangle_shape_icon.png</iconset>
   </property>
   <property name="text">
    <string>DrawRectangle</string>
   </property>
   <property name="toolTip">
    <string>draw a rectangle</string>
   </property>
  </action>
  <action name="actionDrawRoundedCornerRectangle">
   <property name="icon">
    <iconset>
     <normaloff>ico/rectangleCircle_icon.png</normaloff>ico/rectangleCircle_icon.png</iconset>
   </property>
   <property name="text">
    <string>DrawRoundedCornerRectangle</string>
   </property>
   <property name="toolTip">
    <string>Draw a RoundedCornerRectangle</string>
   </property>
  </action>
  <action name="actionDrawTriangle">
   <property name="icon">
    <iconset>
     <normaloff>ico/shape_triangle_icon.png</normaloff>ico/shape_triangle_icon.png</iconset>
   </property>
   <property name="text">
    <string>DrawTriangle</string>
   </property>
   <property name="toolTip">
    <string>Draw a Triangle</string>
   </property>
  </action>
  <action name="actionDrawCircle">
   <property name="icon">
    <iconset>
     <normaloff>ico/circle_oval_shape_icon.png</normaloff>ico/circle_oval_shape_icon.png</iconset>
   </property>
   <property name="text">
    <string>DrawCircle</string>
   </property>
   <property name="toolTip">
    <string>Draw a Circle</string>
   </property>
  </action>
  <action name="actionDrawEllipse">
   <property name="icon">
    <iconset>
     <normaloff>ico/ellipse_adobe illustrator tool_circle_icon.png</normaloff>ico/ellipse_adobe illustrator tool_circle_icon.png</iconset>
   </property>
   <property name="text">
    <string>DrawEllipse</string>
   </property>
   <property name="toolTip">
    <string>Draw a Ellipse</string>
   </property>
  </action>
  <action name="actionDrawPolygon">
   <property name="icon">
    <iconset>
     <normaloff>ico/linearray_lines_shape_icon.png</normaloff>ico/linearray_lines_shape_icon.png</iconset>
   </property>
   <property name="text">
    <string>DrawPolygon</string>
   </property>
   <property name="toolTip">
    <string>Draw a Polygon</string>
   </property>
  </action>
  <action name="actionGrabOnce">
   <property name="icon">
    <iconset>
     <normaloff>ico/grabOnce_icon.png</normaloff>ico/grabOnce_icon.png</iconset>
   </property>
   <property name="text">
    <string>GrabOnce</string>
   </property>
   <property name="toolTip">
    <string>Grab Once</string>
   </property>
  </action>
  <action name="actionGrabContinue">
   <property name="icon">
    <iconset>
     <normaloff>ico/grabContinue_icon.png</normaloff>ico/grabContinue_icon.png</iconset>
   </property>
   <property name="text">
    <string>GrabContinue</string>
   </property>
   <property name="toolTip">
    <string>Grab Continue</string>
   </property>
  </action>
  <action name="actionGrabStop">
   <property name="icon">
    <iconset>
     <normaloff>ico/grabStop_icon.png</normaloff>ico/grabStop_icon.png</iconset>
   </property>
   <property name="text">
    <string>GrabStop</string>
   </property>
   <property name="toolTip">
    <string>Grab Stop</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QxtSpanSlider</class>
   <extends>QSlider</extends>
   <header>QxtSpanSlider.h</header>
  </customwidget>
  <customwidget>
   <class>ExtendQGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>ExtendQGraphicsView.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
